master/internal/resourcemanagers/fitting_methods.go

```diff
@@ -54,11 +51,10 @@ func WorstFit(req *sproto.AllocateRequest, agent *agentState) float64 {
 	switch {
 	case agent.numUsedSlots() != 0 || req.SlotsNeeded != 0:
 		return float64(agent.numEmptySlots()) / float64(agent.numSlots())
-	case agent.maxZeroSlotContainers == 0:
+	case agent.numZeroSlots() == 0:
 		return 0.0
 	default:
-		return float64(agent.maxZeroSlotContainers-agent.numZeroSlotContainers()) /
-			float64(agent.maxZeroSlotContainers)
+		return float64(agent.numEmptyZeroSlots()) / float64(agent.numZeroSlots())
```

Comment:
may be nice to note that this logic only makes sense/works because hard constraints exclude agents that are draining.

Reply:
if it's draining, numEmptyZeroSlots would be zero. so this formula makes sense to me? the hard constraint is already included in it.

Reply:
sorry, i left the comment here but i was thinking it on line 39, where `numUsedSlots() > 0` for draining agents and `numEmptySlots()` is 0 so the affinity is 1 go fit to draining agents, but draining agents are excluded by hard constraints because they dont have enough slots or zero slots to satisfy the task.